client/components/domains/transfer-domain-step/index.jsx

```diff
@@ -204,23 +213,56 @@ class TransferDomainStep extends React.Component {
 		);
 	}
 
+	startPendingInboundTransfer() {
+		const { searchQuery, selectedSite, translate } = this.props;
+		startInboundTransfer( selectedSite.ID, searchQuery, ( error, result ) => {
+			if ( result ) {
+				fetchDomains( selectedSite.ID );
+				page( domainManagementTransferInLink( selectedSite.slug, searchQuery ) );
+			} else {
+				const errorMessage = error.message || translate( 'We were unable to start the transfer.' );
```

Comment:
No need for the error in state. We can just show the disappearing notice, and no changes in render. :)